lib/SILOptimizer/SILCombiner/SILCombinerCastVisitors.cpp

```diff
@@ -581,33 +587,67 @@ SILInstruction *SILCombiner::visitUnconditionalCheckedCastAddrInst(
   //                                ConcreteError in %dest : $*ConcreteError
   //
   // to:
-  //   ...
+  //
   //   retain_value %value : $ConcreteError
+  //   ...
+  //   %box = alloc_existential_box $Error, $ConcreteError
+  //   %a = project_existential_box $ConcreteError in %b : $Error
+  //   store %value to %a : $*ConcreteError
+  //   %err = alloc_stack $Error
+  //   store %box to %err : $*Error
   //   destroy_addr %err : $*Error
   //   store %value to %dest $*ConcreteError
   //
-  // This lets the alloc_existential_box become dead and it can be removed in
-  // following optimizations.
+  // OSSA Pattern:
+  //
+  //   %value = ...
+  //   ...
+  //   %box = alloc_existential_box $Error, $ConcreteError
+  //   %a = project_existential_box $ConcreteError in %b : $Error
+  //   store %value to [init] %a : $*ConcreteError
+  //   %err = alloc_stack $Error
+  //   store %box to [init] %err : $*Error
+  //   %dest = alloc_stack $ConcreteError
+  //   unconditional_checked_cast_addr Error in %err : $*Error to
+  //                                ConcreteError in %dest : $*ConcreteError
+  //
+  // to:
+  //
+  //   %value_copy = copy_value %value
+  //   ...
+  //   %box = alloc_existential_box $Error, $ConcreteError
+  //   %a = project_existential_box $ConcreteError in %b : $Error
+  //   store %value to [init] %a : $*ConcreteError
+  //   %err = alloc_stack $Error
+  //   store %box to [init] %err : $*Error
+  //   destroy_addr %err : $*Error
+  //   store %value to %dest $*ConcreteError
+  //
+  // In both cases, this lets the alloc_existential_box become dead and it can
+  // be removed in other subsequent optimizations.
   SILValue val = getConcreteValueOfExistentialBoxAddr(uccai->getSrc(), uccai);
   while (auto *cvi = dyn_cast_or_null<CopyValueInst>(val))
     val = cvi->getOperand();
   if (canBeUsedAsCastDestination(val, uccai, DA)) {
     // We need to copy the value at its insertion point.
     {
-      auto *valInsertPt = val->getDefiningInsertionPoint();
-      if (!valInsertPt)
+      auto *nextInsertPt = val->getNextInstruction();
+      if (!nextInsertPt)
```

Comment:
I actually don't think we should ever bailout after `getNextInstruction`. Can't we just assert in side `getNextInstruction` that it isn't called on a terminator?